File: backend/src/main/java/fu/osms/auth/service/UserService.java
package fu.osms.auth.service;

import fu.osms.auth.dto.request.UpdateProfileRequest;
import fu.osms.auth.dto.request.UserRequest;
import fu.osms.auth.dto.response.UserProfileResponse;
import fu.osms.auth.dto.response.UserResponse;
import fu.osms.common.dto.PageResponse;

import java.util.UUID;

public interface UserService {

    // ── CRUD ────────────────────────────────────────────────────────────────
    UserResponse create(UserRequest request);

    UserResponse getById(UUID id);

    UserResponse getByEmail(String email);

    PageResponse<UserResponse> getAll(int page, int size);

    UserResponse update(UUID id, UserRequest request);

    void changePassword(UUID id, String oldPassword, String newPassword);

    void delete(UUID id);

    // ── Profile (current user) ────────────────────────────────────────────
    UserProfileResponse getMyProfile(String email);

    UserProfileResponse updateMyProfile(String email, UpdateProfileRequest request);

    UserProfileResponse getProfileById(UUID userId);

    void changePassword(String email, String oldPassword, String newPassword);
}

File: frontend/src/features/user/services/userService.js
import userApi from '../../../api/userApi';

const userService = {
  /**
   * Get current user's profile.
   */
  getMyProfile: async () => {
    const data = await userApi.getMyProfile();
    return data;
  },

  /**
   * Update current user's profile.
   * @param {Object} profileData - { fullName, phone, avatarUrl }
   */
  updateMyProfile: async (profileData) => {
    const data = await userApi.updateMyProfile(profileData);
    return data;
  },

  /**
   * Get user profile by ID.
   */
  getProfileById: async (id) => {
    const data = await userApi.getProfileById(id);
    return data;
  },

  /**
   * Change current user's password.
   * @param {Object} data - { oldPassword, newPassword, confirmPassword }
   */
  changeMyPassword: async (data) => {
    const result = await userApi.changeMyPassword(data);
    return result;
  },
};

export default userService;
